lib/net/dhcp/ztp_record.rb

```diff
@@ -0,0 +1,19 @@
+module Net::DHCP
+  class ZTPRecord < Record
+    attr_accessor :firmware, :vendor
+
+    def initialize(opts = { })
+      super(opts)
+      self.firmware    = opts[:firmware]
+      self.vendor      = opts[:vendor]
+    end
+
+    def attrs
+      super.merge(
+        {
+          :ztp_firmware   => firmware,
+          :ztp_vendor     => vendor
+        }).compact
+    end
```

Comment:
Can you use Ruby memorization here and change this the same way for all record types? We call `attrs` method multiple times during figuring out dupes and this is slow.

Reply:
Hey @lzap , great input! Fixed it and also used "compact" for the other record types.